### PR TITLE
fix(duckdb)!: Generation of exp.SHA2, exp.Transform, exp.IgnoreNulls

### LINKED ISSUE #3972: Incorrect Parsing Between Spark SQL and DuckDB
There are a couple of mistakes when translating from Spark to DuckDB

**Fully reproducible code snippet**
```python
from sqlglot import parse_one
 query = """SELECT SHA2(str, 256), ANY_VALUE(str2, true), FIRST(str3, true), LAST(str4, true), TRANSFORM(array_col)
FROM table 
GROUP BY group_col"""
 parse_one(query, dialect="spark").sql(dialect="duckdb")

``` 
The output is:
```python
SELECT SHA2(str, 256), ANY_VALUE(str2 IGNORE NULLS), FIRST(str3 IGNORE NULLS), LAST(str4 IGNORE NULLS), TRANSFORM(array_col) FROM "table" GROUP BY group_col
``` 
Which fails for three different reasons:
1. SHA2 is not a function in duckdb - the relevant function is sha256 
2. TRANSFORM should be LIST_TRANSFORM
3. When using non-window function, duckdb doesn't support RESPECT/IGNORE NULLS. The default behavior is without NULLs 
The expected output
**Official Documentation**
1. https://duckdb.org/docs/sql/functions/utility.html - SHA256
2. https://duckdb.org/docs/sql/functions/list.html - LIST_TRANSFORM
3. https://duckdb.org/docs/sql/functions/aggregates.html - aggregate functions

### PR DESCRIPTION
**Pull request by VaggelisD**
Fixes #3972

This PR addresses all 3 issues (SHA256, LIST_TRANSFORM, IGNORE NULLS) as the first 2 are trivial to fix.

Docs
----------
[DuckDB Window Functions](https://duckdb.org/docs/sql/functions/window_functions.html) | [DuckDB LIST_TRANSFORM](https://duckdb.org/docs/sql/functions/list.html#list_transformlist-lambda) | [DuckDB SHA256](https://duckdb.org/docs/sql/functions/utility.html#sha256value)

### CODE REVIEW

File: sqlglot/dialects/duckdb.py [hunk 1]
```diff
@@ -41,6 +42,14 @@
     exp.DateAdd, exp.TimeAdd, exp.DatetimeAdd, exp.TsOrDsAdd, exp.DateSub, exp.DatetimeSub
 ]
 
+WINDOW_FUNCS = (
```
Comment (tobymao) [inline sqlglot/dialects/duckdb.py:45]: can we make this a subclass instead? change the expression subclass to window func

Comment (georgesittas) [reply sqlglot/dialects/duckdb.py:45]: I think these are a subset of the actual window funcs, right @VaggelisD? IIRC duckdb only allows `IGNORE NULLS` for certain window funcs, hence this tuple? If so, I'd actually suggest renaming this to something like `WINDOW_FUNCS_WITH_IGNORE_NULLS` or something.

Comment (VaggelisD) [reply sqlglot/dialects/duckdb.py:45]: Right, this was a detail that I should have documented more clearly; DuckDB differentiates the aggregate functions from the general-purpose window functions, and `IGNORE NULLS` is accepted on a subset of the general-purpose window funcs. 

This [table here](https://duckdb.org/docs/sql/functions/window_functions.html#general-purpose-window-functions) documents them, note those with `IGNORE NULLS`. The `WINDOW_FUNCS` set holds this subset, but it's not accurately named as George pointed out, will rename it.

Comment (tobymao) [reply sqlglot/dialects/duckdb.py]: got it

**Review by tobymao — APPROVED** [review]

**Pull request merged**